pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -129,12 +119,13 @@ func (m *KubeletMonitor) scrapeKubelet(node *api.Node) {
 			// It's a valid case if a node is available from the full discovery but not available during sampling discoveries.
 			// Need to wait for a full discovery to fetch the available nodes.
 			glog.Warningf("Failed to get resource metrics summary sample from %s. Waiting for the next full discovery.", node.Name)
-			return
```

Comment:
Shall we keep the old logic and return here instead of continuing?

Reply:
@marvelyue , what do you think? :)

Reply:
Based on the code comment, to be safer, I'm leaning to the old logic by returning here instead of continuing. If node is not available during sampling discovery, then continuing on following logic to parse pod and node stats could lead to unexpected results.

Will take a closer look and revisit the whole logic later today.

Reply:
Updated based on the comments here. Thanks @marvelyue , I know you would notice! :)